hikari/utilities/cache.py

```diff
@@ -368,20 +342,23 @@ class PrivateTextChannelData(BaseData[channels.PrivateTextChannel]):
     last_message_id: typing.Optional[snowflakes.Snowflake] = attr.ib()
     recipient_id: snowflakes.Snowflake = attr.ib()
 
-    @classmethod
-    def get_fields(cls) -> typing.Collection[str]:
-        return "id", "name", "last_message_id"
-
-    def build_entity(
-        self, target: typing.Type[channels.PrivateTextChannel], **kwargs: typing.Any
-    ) -> channels.PrivateTextChannel:
-        return super().build_entity(target, type=channels.ChannelType.PRIVATE_TEXT, **kwargs)
+    def build_entity(self, **kwargs: typing.Any) -> channels.PrivateTextChannel:
```

Comment:
isn't passing a dict each time equally inefficient?

Reply:
we were doing that before and that's the only way to not violate that guy's law about substitution here and annoy mypy

Reply:
Liskov's substitution principle*